blockprotocol/blockprotocol
# Publish block-template and create-block-app

This PR introduces a basic version of `create-block-app`, which simplifies initial onboarding. I have recycled the existing scripts as much as I could, thus only focusing on the NPM migration.

I have already published [`block-template@0.0.2`](https://www.npmjs.com/package/block-template) and [`create-block-app@0.0.2`](https://www.npmjs.com/package/create-block-app) to make sure we have testable package versions before the merge.

## How to test this

```sh
cd /tmp

npx create-block-app hello-world

yarn create block-app hello-world-2 /tmp/random-dir/random-sub-dir
```

Both commands should work both in zsh and PowerShell (tested on macOS). You should see the contents of `block-template` in `/tmp/hello-world` and `/tmp/random-dir/random-sub-dir` after running them. Both `package.json` files will be slightly amended and:
- include your username (from git)
- not include some vendor-specific attributes like homepage or repository.

## Next steps

There are plenty of ways to improve block package metadata and the template itself. Please share your ideas and let’s implement them in follow-up PRs!
